crates/ruff/src/checkers/ast.rs

```diff
@@ -2631,7 +2631,10 @@ where
                 if self.settings.rules.enabled(&Rule::OSErrorAlias) {
                     pyupgrade::rules::os_error_alias(self, &expr);
                 }
-                if self.settings.rules.enabled(&Rule::IsinstanceWithTuple) {
+                if self.settings.rules.enabled(&Rule::IsinstanceWithTuple)
+                    && !self.settings.pyupgrade.keep_runtime_typing
```

Comment:
This flag is redundant, but it's already redundant for the other PEP 604 rule (since it's equivalent to turning off the rule), so I think it makes sense to be consistent.